pkg/test/framework/components/istio/kubealpha.go

```diff
@@ -0,0 +1,241 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package istio
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"sync"
+
+	"github.com/hashicorp/go-multierror"
+
+	"istio.io/istio/pkg/test/deployment"
+	ienv "istio.io/istio/pkg/test/env"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/resource"
+	"istio.io/istio/pkg/test/scopes"
+)
+
+type installerComponent struct {
+	id          resource.ID
+	settings    Config
+	ctx         resource.Context
+	environment *kube.Environment
+}
+
+var _ io.Closer = &installerComponent{}
+var _ Instance = &installerComponent{}
+var _ resource.Dumper = &installerComponent{}
+
+// ID implements resource.Instance
+func (i *installerComponent) ID() resource.ID {
+	return i.id
+}
+
+func (i *installerComponent) Settings() Config {
+	return i.settings
+}
+
+func (i *installerComponent) Close() error {
+	if !i.settings.DeployIstio {
+		return nil
+	}
+	namespaces := GetDeployedNamespaces(i.settings)
+	errchan := make(chan error, len(namespaces))
+	wg := &sync.WaitGroup{}
+	for _, ns := range namespaces {
+		ns := ns
+		wg.Add(1)
+		go func() {
+			err := i.environment.Accessor.DeleteNamespace(ns)
+
+			if err == nil {
+				err = i.environment.Accessor.WaitForNamespaceDeletion(ns)
+			}
+
+			if err != nil {
+				errchan <- err
+			}
+			wg.Done()
+		}()
+	}
+	wg.Wait()
+
+	close(errchan)
+	var err error
+	for e := range errchan {
+		err = multierror.Append(err, e)
+	}
+
+	return err
+}
+
+func (i *installerComponent) Dump() {
+	scopes.CI.Errorf("=== Dumping Istio Deployment State...")
+
+	d, err := i.ctx.CreateTmpDirectory("istio-state")
+	if err != nil {
+		scopes.CI.Errorf("Unable to create directory for dumping Istio contents: %v", err)
+		return
+	}
+	for _, ns := range GetDeployedNamespaces(i.settings) {
+		deployment.DumpPodState(d, ns, i.environment.Accessor)
+		deployment.DumpPodEvents(d, ns, i.environment.Accessor)
+
+		pods, err := i.environment.Accessor.GetPods(ns)
+		if err != nil {
+			scopes.CI.Errorf("Unable to get pods from the system namespace: %v", err)
+			return
+		}
+
+		for _, pod := range pods {
+			for _, container := range pod.Spec.Containers {
+				l, err := i.environment.Logs(pod.Namespace, pod.Name, container.Name, false /* previousLog */)
+				if err != nil {
+					scopes.CI.Errorf("Unable to get logs for pod/container: %s/%s/%s", pod.Namespace, pod.Name, container.Name)
+					continue
+				}
+
+				fname := path.Join(d, fmt.Sprintf("%s-%s.log", pod.Name, container.Name))
```

Comment:
filepath instead of path is logically more correct (even if we don't care about Windows, filepath will make it clear you're constructing a path for use with a real local file-systems and something else that has a concept of paths